### PR TITLE
 feat(list): can specify navigation type

### PR DESCRIPTION
**Pull request by yjlo**
Added a prop to List
## Description

Added `navigationDirection` prop so that when navigating through the list you can choose to use UP/DOWN, LEFT/RIGHT, or both.
## Motivation and Context

Visually a horizontal list, you may not want UP/DOWN to navigate through the list. Same ver vertical list and LEFT/RIGHT

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: react/src/lib/List/index.js [hunk 1]
```diff
@@ -426,6 +443,8 @@ List.propTypes = {
   id: PropTypes.string,
   /** @prop Optional tabType prop type to manually set child role | 'listitem' */
   itemRole: PropTypes.string,
+  /** @prop Restricts the traversal of the list with either UP/DOWN, LEFT/RIGHT, or both | 'both' */
+  navigationDirection: PropTypes.oneOf(['vertical', 'horizontal', 'both']),
```
Comment (bsiboliban-csco) [inline react/src/lib/List/index.js:447]: Is `both` a common value that we use? Seems like `any` or `all` might be a better choice.

File: react/src/lib/List/index.js [hunk 2]
```diff
@@ -174,29 +174,45 @@ class List extends React.Component {
         flag = true;
         break;
 
-      case 38:
-      case 37:
-        this.getNextFocusedChild(items, tgt, -1);
-        this._needsRefocus = true;
-        if(!shouldPropagateKeyDown) flag = true;
+      case 38: // UP
+        if (navigationDirection === 'both' || navigationDirection === 'vertical') {
```
Comment (bsiboliban-csco) [inline react/src/lib/List/index.js:178]: Might consider assigning `navigationDirection === 'both' || navigationDirection === 'vertical'` to a variable to make clear the intent. `isVerticalOrAny`? Same goes for when you're checking for `horizontal`.

File: react/src/lib/List/index.js (hunk 1 above)
Comment (yjlo) [reply react/src/lib/List/index.js:447]: I'm not really sure what we use, I feel like any is too vague and all as well.

File: react/src/lib/List/index.js (hunk 2 above)
Comment (yjlo) [reply react/src/lib/List/index.js:178]: Fair will make appropriate changes.

**Pull request merged**